src/Sparse/ZZRow.jl

```diff
@@ -249,7 +251,7 @@ function sparse_row(R::ZZRing, A::Vector{Tuple{Int64, ZZRingElem}}; sort::Bool =
   return SRow(R, l, a)
 end
 
-function sparse_row(R::ZZRing, pos::Vector{Int64}, val::AbstractVector{ZZRingElem}; sort::Bool = true)
+function sparse_row(R::ZZRing, pos::Vector{Int64}, val::AbstractVector{T}; sort::Bool = true) where T
```

Comment:
This needs to stay `ZZRingElem`, as it uses `ZZRingElem_Array()` in the body!

Reply:
Maybe false alarm for this: the `push!(::ZZRingElem_Array, -)` takes care of other input types, at least I found a version for `Int`. So this change makes `sparse_row(ZZ, ::Vector{Int}, ::Vector{Int})` work while moving errors for unsupported types further down.

Reply:
Yes that was the intention, to get feature parity with the generic `sparse_row(R::Ring, J::Vector{Int}, V::Vector{T})` which explicitly coerces into the ring:
```julia
function sparse_row(R::Ring, pos::Vector{Int}, val::AbstractVector{T}; sort::Bool = true) where T
  if sort
    p = sortperm(pos)
    pos = pos[p]
    val = val[p]
  end
  if T === elem_type(R)
    return SRow(R, pos, val)
  else
    mapval = map(R, val)::Vector{elem_type(R)}
    return SRow(R, pos, mapval)
  end
end
```